Exercise_1.java

```diff
@@ -1,32 +1,46 @@
+package com.s30;
 class Stack { 
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
     boolean isEmpty() 
     { 
-        //Write your code here 
+    	 return top == -1;
     } 
 
     Stack() 
     { 
+    	a = new int[MAX];
+
+    	top = -1;
         //Initialize your constructor 
     } 
   
     boolean push(int x) 
     { 
+    	a[++top] = x;
+    	if(isFull())
+    		return false;
+    	else
+    		return true;
         //Check for stack Overflow
         //Write your code here
     } 
   
     int pop() 
     { 
+    	return a[top--];
         //If empty return 0 and print " Stack Underflow"
         //Write your code here
     } 
   
+    public Boolean isFull() {
+        return top == MAX - 1;
+      }
     int peek() 
     { 
+    	return a[top-- + 1];

```

Comment:
This way you'll lose the top pointer, for peek the top pointer decremented.
"return a[top]" should work just fine.